thirdparty/patches/rules_boost-flyweight.patch

```diff
@@ -0,0 +1,28 @@
+diff --git BUILD.boost BUILD.boost
```

Comment:
Why do we need this patch?

Reply:
bcz the boost bazel version we used, i.e. "https://github.com/nelhage/rules_boost/archive/652b21e35e4eeed5579e696da0facbe8dba52b1f.tar.gz", doesn't include the flyweight module.

Reply:
oh, by saying boost version I meant the `rules_boost`. I had tried update the `rule_boost` to the version that includes the `flyweight`, but that version has conflicts with our `windows-linkopts` patch. I was worried that a newer `rule_boost` may further bring more latent problems.

Reply:
Do you mind spending some time to fix the `windows-linkopts` patch? I think eventually we need to be able to update the `rule_boost` version.

We can make sure the upgrade doesn't break things by looking at CI results.